jenkins/deploy/Jenkinsfile

```diff
@@ -0,0 +1,80 @@
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "${JDK_VERSION}" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "${MAVEN_VERSION}"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sh """
+                            mvn clean install -DskipITs
+                        """
+                    }
+                }
+            }
+        }
+
+        stage("Deploy Primary") {
+            steps {
+                configFileProvider([configFile(fileId: "iis-build-properties", variable: 'BUILD_PROPERTIES')]) {
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "${JDK_VERSION}" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "${MAVEN_VERSION}"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sshagent(['iis-deployment-credentials']) {
+                            sh """
+                                cd iis-wf/iis-wf-primary && \
+                                mvn clean package -Poozie-package,deploy \
+                                    -Dworkflow.source.dir=eu/dnetlib/iis/wf/primary/main \
+                                    -DiisConnectionProperties=${WORKSPACE}/connection-properties/iis-cdh5-test-gw.properties
+                            """
+                        }
+                    }
+                }
+            }
+        }
+
+        stage("Release Primary") {
+            steps {
+                script {
+                    def propertiesContent = readFile("${WORKSPACE}/connection-properties/iis-cdh5-test-gw.properties")
+                    def properties = [:]
+                    propertiesContent.split("\\n").each { line ->
+                        if (line.contains('=')) {
+                            def nameAndValue = line.split('=')
+                            properties[nameAndValue[0]] = nameAndValue[1]
+                        }
+                    }
+                    env.USERNAME = properties['iis.hadoop.frontend.user.name']
+                    env.HOSTNAME = properties['iis.hadoop.frontend.host.name']
+                }
+                sshagent(['iis-deployment-credentials']) {
+                    sh """
+                        WORKING_DIR=/tmp/${USERNAME}/`date +%Y-%m-%d`/apps/primary && \
+                        ssh ${USERNAME}@${HOSTNAME} "\
+                            rm -rf \${WORKING_DIR} && \
+                            mkdir -p \${WORKING_DIR}
+                            " && \
+                        scp -r ${WORKSPACE}/jenkins/deploy/apps/primary/* ${USERNAME}@${HOSTNAME}:\${WORKING_DIR} && \
+                        ssh ${USERNAME}@${HOSTNAME} "\
+                            cd \${WORKING_DIR} && \
+                            sh deploy_release.sh
+                            "
+                    """
```

Comment:
There are several things here that I am not sure if they would be improvements:
- The multiple levels of quoting, escaping and variable substitution confused me a bit - maybe use ''' at toplevel to simplify?
- Most of the backslashes at the line ends should be unnecessary
- The script is run by Jenkins with `set -xe` so the toplevel  `&&` do not really change what gets executed on failure, maybe we can get rid of them, or is it too unobvious?
- Can we assume that `$WORKSPACE` will never contain a space, or is it better to quote it just in case?

```suggestion
                    sh '''
                        WORKING_DIR=/tmp/${USERNAME}/`date +%Y-%m-%d`/apps/primary
                        ssh ${USERNAME}@${HOSTNAME} "
                            rm -rf ${WORKING_DIR} && 
                            mkdir -p ${WORKING_DIR}
                            "
                        scp -r "${WORKSPACE}"/jenkins/deploy/apps/primary/* ${USERNAME}@${HOSTNAME}:${WORKING_DIR}
                        ssh ${USERNAME}@${HOSTNAME} "
                            cd ${WORKING_DIR} &&
                            sh deploy_release.sh
                            "
                    '''
```

Reply:
I've tested all the proposed changes and it works, so I'm committing this implementation :)